app/services/articles/detect_animated_images.rb

```diff
@@ -0,0 +1,36 @@
+module Articles
+  module DetectAnimatedImages
+    IMAGES_IN_LIQUID_TAGS_SELECTORS = [
+      ".liquid-comment img", # CommentTag
+      ".ltag-github-readme-tag img", # GithubReadmeTag
+      ".ltag__link__servicename img", # MediumTag
+      ".ltag__link__taglist img", # LinkTag
+      ".ltag__reddit--container img", # RedditTag
+      ".ltag__stackexchange--container img", # StackexchangeTag
+      ".ltag__twitter-tweet img", # TweetTag
+      ".ltag__user img", # UserTag and OrganizationTag
+      ".ltag__user-subscription-tag img", # UserSubscriptionTag
+      ".ltag_github-liquid-tag img", # GitHubIssueTag
+      ".podcastliquidtag img", # PodcastTag
+    ].join(", ").freeze
+
+    def self.call(article)
```

Comment:
Is there a reason not to put this functionality inside the worker directly?

Reply:
Most of our workers are coded this way: very little logic calling a service :) It's also a bit more future proof, in the sense that if this service starts becoming more complicated, it can be evolved separately from the worker itself

Reply:
Yeah, I think discussing it is very likely beyond the scope of this PR, but the idea behind the question is that we can run workers synchronously. 🙂

Reply:
Sure, we could, we do in [some places](https://github.com/forem/forem/search?q=new.perform). `new.perform` is a bit clunky though we could add a `.call` class method in a base class and have all workers inherit from that. 

That said there are a few reasons to try keeping services and workers separated:

- workers are coupled to Sidekiq (they literally include the `Sidekiq::Worker` module)
- the word `worker`, although overloaded, makes me think of something that runs externally, not something that could work in process and out of process at the same time
- separating the two means you can limit the worker to contain worker related logic (loading the objects from the serialized arguments, retrying, raise exceptions and so on)

The downside is that looking at a worker doesn't show you the whole story, but that's what abstraction is in theory, if things are named correctly, they should be self explanatory:

```ruby
module Articles
  class BustCacheWorker < BustCacheBaseWorker
    def perform(article_id)
      article = Article.find_by(id: article_id)
      return unless article

      EdgeCache::BustArticle.call(article)
    end
  end
end
```

Reply:
> but the idea behind the question is that we can run workers synchronously. 

There was a whole discussion on Ruby Twitter about this not too long ago, e.g. are service objects even needed, or can workers fully replace them since they already provide interfaces for both sync and async execution? 

Personally, I fall in the same camp as rhymes, i.e. I prefer a PORO that contains the actual logic which then gets used in different contexts that deal with the transport layer/execution context (a controller, job, rake tasks, etc.).